components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/dao/CacheBackedIdPMgtDAO.java

```diff
@@ -278,6 +278,50 @@ public IdentityProvider getIdPByResourceId(String resourceId, int tenantId, Stri
         return identityProvider;
     }
 
+    /**
+     * Get the updated IDP details using the resource ID.
+     *
+     * @param resourceId   resource ID of the identity provider.
+     * @param tenantId     Tenant ID of the identity provider.
+     * @param tenantDomain Tenant domain of the identity provider.
+     * @return Updated identity provider with given resource ID.
+     * @throws IdentityProviderManagementException
+     */
+    public IdentityProvider getUpdatedIdPByResourceId(String resourceId, int tenantId, String tenantDomain) throws
+            IdentityProviderManagementException {
+
+        IdentityProvider identityProvider;
+        IdPResourceIdCacheKey cacheKey = new IdPResourceIdCacheKey(resourceId);
+        IdPCacheEntry entry = idPCacheByResourceId.getValueFromCache(cacheKey, tenantDomain);
+
+        if (entry != null) {
+            /*
+            Before updating an IDP we clear the cache. Therefore, if we get a cache hit again at this point, it should
+            be a reason of some other process done on the same IDP. Hence, we need to clear the cache before proceed
+            in order to generate a correct response.
+             */
+            if (log.isDebugEnabled()) {
+                log.debug("Cache entry found for Identity Provider with resource ID: " + resourceId
+                        + ". Hence clear the cache before proceed.");
+            }
+            identityProvider = entry.getIdentityProvider();
+            clearIdpCache(identityProvider.getIdentityProviderName(), identityProvider.getResourceId(),
+                    tenantId, tenantDomain);
+        }
+
+        identityProvider = idPMgtDAO.getIDPbyResourceId(null, resourceId, tenantId, tenantDomain);
+
+        if (identityProvider != null) {
+            addIdPCache(identityProvider, tenantDomain);
+        } else {
+            if (log.isDebugEnabled()) {
+                log.debug(String.format("No IDP found with resource ID: %s in DB", resourceId));
+            }
+        }
```

Comment:
Better if we can use `If without else` here. Not nice to have a separate else block just for a debug log.

Reply:
We can return the method if `if (identityProvider == null) {` with a log. WDYT?